integr8ly/integreatly-operator
# MGDAPI-6015 change 3scale to add topologySpreadConstrants via apimanager CR

# Issue link

https://issues.redhat.com/browse/MGDAPI-6015

# What

when the mas-0.11.8 build comes out this needs to be merged

# Verification steps




- [ ] cherry-pick this commit to the mas-0.11.8 branch 
- [ ] Install rhoam locally
```bash
INSTALLATION_TYPE=managed-api LOCAL=false make cluster/prepare/local
INSTALLATION_TYPE=managed-api LOCAL=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
INSTALLATION_TYPE=managed-api LOCAL=false make code/run
```
- [ ] Wait for the install to finish
- [ ] Check that the topologySpreadConstraint has been added to all the DC
```bash
for dc in $(oc get deploymentconfigs --namespace=redhat-rhoam-3scale | grep config | awk '{print $1}'); do echo $dc "===================================>"; oc get dc $dc -oyaml | yq '.spec.template.spec.topologySpreadConstraints' ; done
```
expected output
```bash
apicast-production ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
apicast-staging ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-cron ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-listener ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-worker ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-app ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-memcache ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-searchd ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-sidekiq ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync-database ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync-que ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
```
